pyproject.toml

```diff
@@ -1,3 +1,38 @@
+[tool.pytest.ini_options]
+minversion = 6.0
+addopts = """
+    -ra
+    --import-mode=importlib
+"""
+
+python_files = [
+    "test_*.py",
+    "*_test.py",
+    "*Test.py"
+]
+
+python_classes = ["*Test"]
+
+testpaths = [
+    "tests/unit",
+    "tests/integration"
+]
+
+# Ignore DeprecationWarnings in external libraries (because we can't do anything about them)
+filterwarnings = [
+    "error",
+    "ignore::DeprecationWarning:dns.*",
+    "ignore::DeprecationWarning:eventlet.*",
```

Comment:
As we don't use eventlet here, it would make to remove this line in at this step.